torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1189,17 +1218,43 @@ TensorExprKernel::TensorExprKernel(const Graph& subgraph) {
         }
       }
     }
+    if (hasRandom_ && hasBroadcast_) {
+      throw std::runtime_error(
```

Comment:
Within reason, I'd like to use a return value to indicate that the fusion failed, instead of through an exception. 

An exception within a debugger makes it hard if someone is trying to get to the real exceptions. Also it is slow enough, if we are autotuning through many possibilities, it might make the whole thing thousands of times slower.

Reply:
I'd like to come back to this at some point.  I want the try/catch for general robustness purposes; I can see using an error code here instead but I want to think through the error codes properly.